lib/websocket_proxy.rb

```diff
@@ -0,0 +1,74 @@
+class WebsocketProxy
+  attr_reader :env, :url, :error
+
+  def initialize(env, console)
+    @env = env
+    @id = SecureRandom.uuid
+    @console = console
+
+    secure = Rack::Request.new(env).ssl?
+    scheme = secure ? 'wss:' : 'ws:'
+    @url = scheme + '//' + env['HTTP_HOST'] + env['REQUEST_URI']
```

Comment:
prefer using a URI builder over string manipulation, if possible (You can probably use `URI::Generic.new(:scheme => scheme ...`)